assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+
+In Eclipse, `debug` logging to the console tends to cause congestion with a score calculation speeds above 10000 per second.
+Neither IntelliJ or the Maven command line suffer from this problem.
+====
+
+. Procedure
+COMMENT: How do I turn on logging?
+
+For example, set the logging level to `debug` logging to see when the phases end and how fast steps are taken:
+
+[source,options="nowrap"]
+----
+INFO  Solving started: time spent (3), best score (-4init/0), random (JDK with seed 0).
+DEBUG     CH step (0), time spent (5), score (-3init/0), selected move count (1), picked move (Queen-2 {null -> Row-0}).
+DEBUG     CH step (1), time spent (7), score (-2init/0), selected move count (3), picked move (Queen-1 {null -> Row-2}).
+DEBUG     CH step (2), time spent (10), score (-1init/0), selected move count (4), picked move (Queen-3 {null -> Row-3}).
+DEBUG     CH step (3), time spent (12), score (-1), selected move count (4), picked move (Queen-0 {null -> Row-1}).
+INFO  Construction Heuristic phase (0) ended: time spent (12), best score (-1), score calculation speed (9000/sec), step total (4).
+DEBUG     LS step (0), time spent (19), score (-1),     best score (-1), accepted/selected move count (12/12), picked move (Queen-1 {Row-2 -> Row-3}).
+DEBUG     LS step (1), time spent (24), score (0), new best score (0), accepted/selected move count (9/12), picked move (Queen-3 {Row-3 -> Row-2}).
+INFO  Local Search phase (1) ended: time spent (24), best score (0), score calculation speed (4000/sec), step total (2).
+INFO  Solving ended: time spent (24), best score (0), score calculation speed (7000/sec), phase total (2), environment mode (REPRODUCIBLE).
+----
+
+All time spent values are in milliseconds.
+
+Everything is logged to http://www.slf4j.org/[SLF4J], which is a simple logging facade
+that delegates every log message to Logback, Apache Commons Logging, Log4j, or java.util.logging.
+Add a dependency to the logging adaptor for your logging framework of choice.
+
+=== Using Logback to log OptaPlanner solver activity
+Logback is the recommended logging frameworkd to use with {PLANNER_SHORT}. Use Logback to log OptaPlanner solver activity.
+
+.Prerequisites
+* You have an OptaPlanner project.
+
+.Procedure
+
+. Add the following Maven dependency to your {PLANNER_SHORT} project's `pom.xml` file:
++
+NOTE: You do not need to add an extra bridge dependency.
++
+[source,xml,options="nowrap"]
+----
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <version>1.x</version>
+    </dependency>
+----
+
+. Configure the logging level on the `org.optaplanner` package in your `logback.xml` file as shown in the following example where `_LEVEL_` is a logging level listed in XREF:
++
+[source,xml,options="nowrap"]
+----
+<configuration>
+
+  <logger name="org.optaplanner" level="_LEVEL_"/>
+
+  ...
+
+</configuration>
+----
```

Comment:
is the section starting with "If instead, you are still using Log4J 1.x (and you do not want to switch to its faster successor, Logback), add the bridge dependency:"
skipped intentionally?